### ISSUE TITLE
Trilinos build failures using Sierra build process

### ISSUE THREAD

**Issue by ajpowelsnl** [trilinos/Trilinos#4742]
Hello,

Sierra-related Trilinos builds are failing on some engineering workstations (ews), and we have not been able to discern the cause.   We have performed bake distributed builds with both gcc and intel compilers.  I can provide complete logs.  Errors are of this type:

make[2]: Leaving directory `/sierra/dev/ajpowel/code_032619/objs/tpls/Trilinos/20c07de8996d4f55'
[ 94%] Built target muelu
make[1]: Leaving directory `/sierra/dev/ajpowel/code_032619/objs/tpls/Trilinos/20c07de8996d4f55'
make: *** [all] Error 2
WARNING: Trilinos build failed!
Rebuilding Makefile (Option change(s): ['--installdir=/sierra/dev/ajpowel/code_032619/objs/tpls/trilinos_tpls/20c07de8996d4f55', '/tpl/trilinos//install-trilinos-tpls']->['--bin-dir=/sierra/dev/ajpowel/code_032619/bin'])
Building bjam...
Using dependency graph:  /sierra/dev/ajpowel/code_032619/bakefiles/bakefile_e3b0c44298fc1c14_20c07de8996d4f55_deps
Using Trilinos out of: /sierra/dev/ajpowel/code_032619/objs/tpls/trilinos_tpls/20c07de8996d4f55
INFO: Changing version for trilinos from dev to external
INFO: Changing version for trilinos-kokkoscore from dev to external
INFO: Changing version for trilinos-kokkoscontainers from dev to external
INFO: Changing version for trilinos-kokkosalgorithms from dev to external
INFO: Changing version for trilinos-tpetraclassic from dev to external
INFO: Changing version for trilinos-tpetracore from dev to external
INFO: Changing version for trilinos-kokkoskernels from dev to external
INFO: Changing version for trilinos-tpetratsqr from dev to external
error: Unable to find file or target named
error:     '/sierra/dev/ajpowel/code_032619/objs/tpls/trilinos_tpls/20c07de8996d4f55/lib/libkokkoscore.a'
error: referred from project at
error:     '/sierra/dev/ajpowel/code_032619/TPLs_src/Nbtools/Trilinos/external/KokkosCore'

Bjam failed!

 
Many thanks!

Best,

AJP

@trilinos/<teamName>

## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of

**Comment by jhux2** [comment]
@srajama1 Not sure who to notify for this.

**Comment by srajama1** [comment]
Adding the other product leads @jwillenbring @kddevin @rppawlo @mperego and Sierra developers  @prwolfe @mhoemmen to see if some one has access to this environment.

**Comment by srajama1** [comment]
@ajpowelsnl I am trying to evaluate this so we can help you. Is this a new failure on a build/configuration that used to work or is this is a new setup ? The error is not very descriptive. Is there a verbose mode in bjam that you can use so we can see more output on where things are going wrong ?

**Comment by srajama1** [comment]
Adding @rrdrake as well.

**Comment by ajpowelsnl** [comment]
Hi there,

Many thanks for your response.  This is a new problem, apparently affecting only Mark Hamilton and me (on our EWS) for “bake” (default) distributed builds.  I have complete log files, but I’m having trouble emailing them, but they’re being bounced b/c of line count, I think.

How can I get them to you?

**Comment by mhoemmen** [comment]
Sierra DevOps has been working on this for a day or two.

**Comment by srajama1** [comment]
@mhoemmen : Can we wait on the Trilinos side for Sierra DevOps to give us more info ? We could also see if a Trilinos framework person can help (I am assuming this is framework related), but I doubt they want to deal with bjam. @jwillenbring : What do you think ?

@ajpowelsnl : I see you are in SNL. You can e-mail it the logs to me srajama@...

**Comment by mhoemmen** [comment]
Correction: @ajpowelsnl and the other Sierra DevOps people have been working on this for a day or two :-) (sorry I didn't recognize your handle!).

**Comment by prwolfe** [comment]
Hmmm, this is short on details, but I just ran into an issue when using a recent master checkout of Trilinos. 

Basically lines 2472 and 2478 in the file packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp are using a const instance of a class (Kokkos::TeamPolicy) to call a const function for a non-const instance. Being named a "set" function I tried removing the "const" from the call in Ifpack2_BlockTriDiContainer_impl.hpp and got a clean build.

The $6 question is how did this get through Trilinos testing and how do we fix that.  This is a very simple issue which should have been caught early.
